server/src/main/java/org/opensearch/common/util/FeatureFlags.java

```diff
@@ -67,6 +67,11 @@ public class FeatureFlags {
      */
     public static final String PLUGGABLE_CACHE = "opensearch.experimental.feature.pluggable.caching.enabled";
 
+    /**
+     * Gates the functionality of integrating protobuf within search API and node-to-node communication.
+     */
+    public static final String PROTOBUF = "opensearch.experimental.feature.search_with_protobuf.enabled";
```

Comment:
Should we call this something more generic since we may be adding more than search and gate all of protobuf transport behind it? What do you think about `"opensearch.experimental.feature.transport.protobuf.enabled"`?

Reply:
Currently, only search related transport will be gated behind this since when we add the transport specific node to node protobuf structure (in the next PR), it will consider only search related classes for now. We can keep adding more classes once we expand the use case. That message would be https://github.com/opensearch-project/OpenSearch/pull/11910/files#diff-3b6cef54e769860b6ff96d7052208f6fe2ef8a86ea04aec232044984ba52e5db.